### ISSUE TITLE
Numeric strings in schematics options

### ISSUE THREAD

**Issue by dgp1130** [angular/angular-cli#24360]
### Command

generate

### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

From http://b/196605664.

Apparently you can't pass in a numeric value into a string argument in schematics. This appears to be interpreted as number and leads to a validation error.

### Minimal Reproduction

```
$ schematics run test --stringValue "1234"
Schematic input does not validate against the Schema: {"stringValue":1234}
Errors:

  Data path ".stringValue" should be string.
```

Minimist seems to interpret "1234" as a number instead of a string, and schematics treats that as a type mismatch.

### Your Environment

```text
See http://b/196605664.
```

**Comment by alan-agius4** [comment]
In schematics CLI we don’t really configure the parser based on the schematics schema.

We could probably port this logic from the Angular CLI. (Maybe have the logic in a shared `ts_libraary`) when the legacy build system is removed.

**Comment by dgp1130** [comment]
Yeah, I know `minimist` doesn't work that way. We'd probably have to change argument parser to actually declare specific options with specific types based on the schematics schema for the ideal solution. Not sure if there's a simpler fix worth exploring, but it's a relatively minor bug so I'm not too worried.

**Comment by alan-agius4** [comment]
Just for the record we no longer use minimist. We use yargs-parser.

**Comment by alan-agius4** [comment]
Also related to https://github.com/angular/angular-cli/issues/16320

We could combine these issues.
